src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs

```diff
@@ -50,12 +50,21 @@ public LargeArrayBuilder(bool initialize)
         /// Use <see cref="Add"/> if adding to the builder is a bottleneck for your use case.
         /// Otherwise, use <see cref="SlowAdd"/>.
         /// </remarks>
+        public void Add(T item) => Add(item, limit: int.MaxValue);
+
+        /// <summary>
+        /// Adds an item to this builder with the specified limit.
+        /// </summary>
+        /// <param name="item">The item to add.</param>
+        /// <param name="limit">The maximum capacity to allocate.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void Add(T item)
+        public void Add(T item, int limit)
```

Comment:
Why is `limit` passed on every call to `Add`? Is it expected to change? If not, shouldn't it be passed in the constructor and then stored in a field? Or is the cost of one more field too high?